web/src/app/alerts/components/AlertDetails.js

```diff
@@ -423,8 +424,18 @@ function AlertDetails(props) {
 
   const { data: alert } = props
 
+  const notices = alert.pendingNotifications.map((n) => ({
+    type: 'WARNING',
+    message: `Notification Pending for ${n.destination}`,
+    details:
+      'One or more notifications are waiting to be sent. Contact methods may be rate limited on a per-service basis.',
```

Comment:
I think we can simplify this a bit. My reasoning comes from the writing guide: https://material.io/design/communication/writing.html#principles

We can probably leave out that one or more notifications are waiting since it's mainly repeating the title.

```suggestion
    message: 'Notification Pending',
    details: `${n.destination} is being temporarily rate-limited.`,
```

Reply:
"pending" messages do not mean rate-limiting necessarily. "waiting to be sent" is about as accurate as it can be -- slow network, service disruption, retries, engine not running, and any number of other things can contribute to a message being in a pending state

Reply:
```suggestion
    message: `Notification Pending for ${n.destination}`,
    details:
      'One or more notifications are waiting to be sent.',
```

Reply:
We could add something to give a little extra context, as long as it's accurate and doesn't imply the reason is known:

Something like:
> This could be due to rate-limiting, processing, or network delays.

The vast majority of users will never see this warning unless something **other than** rate-limiting is happening (e.g., network delays, prioritization for non-first-alerts, service disruption, etc...). Those that are rate-limiting are a smaller subset of configurations, and thus the same users/services each time (e.g., a service that alerts frequently, or too-aggressive notification settings).

In any case, since rate-limiting has a small inference that the user did something wrong or should configure differently, we should avoid any wording that implies the user is being rate-limited unless it's 100% accurate -- and until we refactor the message-sending code with time-to-send & reason, we do not have that information available.

The goal here is only to inform the user that there is a message ready, but not yet in the alert log, so they know the policy is processing as it should.